lib/iiif_manifest/manifest_builder/record_property_builder.rb

```diff
@@ -19,6 +20,30 @@ def apply(manifest)
       def viewing_hint
         (record.respond_to?(:viewing_hint) && record.send(:viewing_hint))
       end
+
+      # If there is a manifest_metadata method, check that it is an Array containing Hashes
+      #  with pairs of 'label' and value'
+      #
+      # @return Array metadata array of label/value hashes
+      def valid_metadata?
+        return unless record.respond_to?(:manifest_metadata)
+        metadata = record.manifest_metadata
+        valid_metadata_structure?(metadata) && valid_metadata_content?(metadata)
+      end
+
+      # Manifest Metadata must be an array containing hashes
+      #
+      # @return Boolean
+      def valid_metadata_structure?(metadata)
+        metadata.is_a?(Array) && metadata[0].is_a?(Hash)
```

Comment:
should we only check the first value or should we check all? e.g.: `metadata.all? { |v| v.is_a?(Hash) }`